src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+
+import {IDelegateRegistry} from './interfaces/IDelegateRegistry.sol';
+import {IVeToken} from './interfaces/IVeToken.sol';
+import {IWardenBoost} from './interfaces/IWardenBoost.sol';
+import {Common} from './Common.sol';
+
+// @author Llama
+abstract contract VeTokenManager is Common {
+  event BuyBoost(address delegator, address receiver, uint256 amount, uint256 duration);
+  event ClaimBoostRewards();
+  event DelegateUpdate(address indexed oldDelegate, address indexed newDelegate);
+  event LockVEBAL(uint256 cummulativeTokensLocked, uint256 lockHorizon);
+  event RemoveBoostOffer();
+  event SellBoost(
+    uint256 pricePerVote,
+    uint64 duration,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  );
+  event SetLockDuration(uint256 newDuration);
+  event SetSpaceId(bytes32 id);
+  event UnlockVEBAL(uint256 tokensUnlocked);
+
+  error MaxFeeExceeded();
+
+  IDelegateRegistry public constant DELEGATE_REGISTRY =
+    IDelegateRegistry(0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446);
+
+  address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+  address public constant B_80BAL_20WETH = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+  address public constant VE_BAL = 0xC128a9954e6c874eA3d62ce62B468bA073093F25;
+  address public constant WARDEN_VE_BAL = 0x42227bc7D65511a357c43993883c7cef53B25de9;
+
+  /// @notice the snapshot delegate for the deposit
+  address public delegate;
+  /// @notice the keccak encoded spaceId of the snapshot space
+  bytes32 public spaceIdBalancer;
+  /// @notice The lock duration of veToken
+  uint256 public lockDurationVEBAL;
+
+  /// @notice Duration in weeks (ie: 1 for 1 Week)
+  /// @param delegator Address of user to buy boost from
+  /// @param receiver Address receiving the boost
```

Comment:
Shouldn't the receiver be the DAO treasury all the time?

Reply:
i was told they wanted flexibility to buy boost for other contracts as well, but we can hard code as address(this) too.
i'll ask in the group chat to dydy